### PR TITLE
IBP-4434 implement names resource calls

### PR DESCRIPTION
**Pull request by cuenyad**
Please, review.

### CODE REVIEW

File: src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	public Name getPreferredName(final Integer gid) {
+		final List<Name> names = daoFactory.getNameDao().getByGIDWithListTypeFilters(gid, 1, null);
+		if (!names.isEmpty()) {
+			return names.get(0);
+		}
+		return null;
+	}
+
+	@Override
+	public UserDefinedField getNameType(final String nameTypeCode) {
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameServiceImpl.java:41]: In GermplasmService we already have
`	List<GermplasmNameTypeDTO> filterGermplasmNameTypes(Set<String> codes);`
You can use this one so we avoid code duplication

File: src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameService.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package org.generationcp.middleware.api.germplasm;
+
+import org.generationcp.middleware.domain.germplasm.GermplasmNameRequestDto;
+import org.generationcp.middleware.pojos.Name;
+import org.generationcp.middleware.pojos.UserDefinedField;
+
+public interface GermplasmNameService {
+
+	Name getNameByNameId(Integer nameId);
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameService.java:9]: getNameById

File: src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameService.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+	Name getPreferredName(Integer gid);
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameService.java:11]: Remove this function and create it in NameDao directly

File: src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+import org.generationcp.middleware.pojos.UserDefinedField;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Collections;
+import java.util.List;
+
+@Service
+@Transactional
+public class GermplasmNameServiceImpl implements GermplasmNameService {
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameServiceImpl.java:18]: Please implement Integration test

**Review by clarysabel — CHANGES_REQUESTED** [review]

File: src/main/java/org/generationcp/middleware/domain/germplasm/GermplasmNameRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.pojomatic.Pojomatic;
+import org.pojomatic.annotations.AutoProperty;
+
+@AutoProperty
+public class GermplasmNameRequestDto {
+
+	@JsonIgnore
+	private Integer id;
```
[the remark this answers is not in the document]
Comment (clarysabel) [reply src/main/java/org/generationcp/middleware/domain/germplasm/GermplasmNameRequestDto.java:11]: same with nameId ?

File: src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameService.java (hunk 3 above)
Comment (cuenyad) [reply src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameService.java]: done.

File: src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameService.java (hunk 2 above)
Comment (cuenyad) [reply src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameService.java]: done.

File: src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameService.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+	UserDefinedField getNameType(final String nameTypeCode);
+
+	void deleteName(Integer nameId);
+
+	void updateName(final GermplasmNameRequestDto germplasmNameRequestDto);
+
+	Integer createName(final GermplasmNameRequestDto germplasmNameRequestDto, final Integer userid);
```
[the remark this answers is not in the document]
Comment (cuenyad) [reply src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameService.java]: done.

File: src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+	private final DaoFactory daoFactory;
+
+	public GermplasmNameServiceImpl(final HibernateSessionProvider sessionProvider) {
+		this.daoFactory = new DaoFactory(sessionProvider);
+	}
+
+	@Override
+	public Name getNameByNameId(final Integer nameId) {
```
[the remark this answers is not in the document]
Comment (cuenyad) [reply src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameServiceImpl.java]: done

File: src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+		return daoFactory.getNameDao().getById(nameId);
+	}
+
+	@Override
+	public Name getPreferredName(final Integer gid) {
```
[the remark this answers is not in the document]
Comment (cuenyad) [reply src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameServiceImpl.java]: done

File: src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameServiceImpl.java (hunk 1 above)
Comment (cuenyad) [reply src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameServiceImpl.java]: done.

File: src/main/java/org/generationcp/middleware/domain/germplasm/GermplasmNameRequestDto.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (cuenyad) [reply src/main/java/org/generationcp/middleware/domain/germplasm/GermplasmNameRequestDto.java]: done.

**Pull request merged**